qiskit/pulse/transforms/channel_transforms.py

```diff
@@ -0,0 +1,330 @@
+
+import numpy as np
+
+from qiskit.pulse import channels
+from qiskit.providers.basebackend import BaseBackend
+from qiskit import pulse, circuit
+from qiskit.pulse.instructions import Instruction
+
+from base_transforms import target_qobj_transform
+from pulse_instruction_types import PhaseFreqTuple, PulseInstruction, OpaqueShape
+from device_info import OpenPulseBackendInfo
+
+InstructionSched = Union[Tuple[int, Instruction], Instruction]
+ScheduleLike = Union[pulse.Schedule, pulse.ScheduleBlock, 
+                     InstructionSched, Iterable[InstructionSched]]
+
+class ParsedInstruction:
+    def __init__(self, 
+                 t0: int,
+                 dt: float,
+                 frame: PhaseFreqTuple, 
+                 data: Union[Instruction, List[Instruction]], 
+                 is_opaque: bool, 
+                 xdata: np.ndarray = None,
+                 ydata: np.ndarray = None):
+        self.t0 = t0
+        self.dt = dt
+        self.frame = frame
+        self.data = data
+        self.is_opaque = is_opaque
+        self.xdata = xdata
+        self.ydata = ydata
+
+class ChannelTransforms:
+    """Channel transform manager."""
+
+    _waveform_group = (pulse.Play, pulse.Delay, pulse.Acquire)
+    _frame_group = (pulse.SetFrequency, pulse.ShiftFrequency,
+                    pulse.SetPhase, pulse.ShiftPhase)
+
+    def __init__(
+        self,
+        waveforms: Dict[int, pulse.Instruction],
+        frames: Dict[int, List[pulse.Instruction]],
+        channel: pulse.channels.Channel,
+        dt: Optional[int] = None,
+        init_phase: Optional[float] = None,
+        init_frequency: Optional[float] = None
+    ):
+        """Creates a new channel transform manager.
+
+        Args:
+            waveforms: List of waveforms shown in this channel.
+            frames: List of frame change type instructions shown in this channel.
+            channel: Channel object associated with this manager.
+        """
+        self._waveforms = waveforms
+        self._frames = frames
+        self.channel = channel
+
+        # initial frame
+        self._init_phase = init_phase or 0
+        self._init_frequency = init_frequency or 0
+
+        # time resolution
+        self._dt = dt or 0
+
+    @classmethod
+    def load_program(cls, 
+                     program: ScheduleLike,
+                     channel: pulse.channels.Channel,
+                     backend: Optional[BaseBackend] = None
+                     ) -> 'ChannelTransforms':
+        """Loads a pulse program represented by ``Schedule``.
+
+        Args:
+            program: Target ``Schedule`` to visualize.
+            channel: The channel managed by this instance.
+            backend: The device to extract parameters from.
+
+        Returns:
+            The channel transform manager for the specified channel.
+        """
+        # flatten the schedule
+        program = target_qobj_transform(program)

```

Comment:
We can assume this is externally done by writing another transform function including this class like a pass. Otherwise this transform will have unnecessary dependency on `target_qobj_transform`.